12-http/example3/main.go

```diff
@@ -18,20 +21,46 @@ type user struct {
 	Phone string
 }
 
-// users is a slice of users.
-var users []user
+var (
+	// mu protects users.
+	mu sync.RWMutex
+	// initialize users, so that the output JSON won't be null
+	// prior to adding users.
+	users = make([]user, 0)
+)
```

Comment:
There aren't a lot of stdlib examples of global mutexes, of course, therefore there won't be any specific stylistic convention of comments vs global var groups involving mutexes. Having the data and the mutex as globals is flakely and artificial anyway.

I'll rework it to use a type involving a mutex (for which there are numerous stylistic precedents I can follow), and define methods on that type to act as the handlers. It'll use about the same amount of code, but provide a good example of building your own `http.Handler`. It'll also reinforce method values that we saw in the methods lesson, since I'll be using some methods as `http.HandlerFunc` values.